### ISSUE TITLE
Segfault when using -y option

### ISSUE THREAD

**Issue by CyberTailor** [vaeth/eix#89]
```
$ eix --version
0.35.1
$ gdb --args eix -y something
GNU gdb (Gentoo 10.2 vanilla) 10.2
<...>
Reading symbols from eix...
(gdb) run
Starting program: /usr/bin/eix -y something
[Thread debugging using libthread_db enabled]
Using host libthread_db library "/lib64/libthread_db.so.1".

Program received signal SIGSEGV, Segmentation fault.
std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >::_M_set_length (__n=0, this=0x555555bbc850)
    at /usr/lib/gcc/x86_64-pc-linux-gnu/11.1.0/include/g++-v11/bits/basic_string.h:217
217             traits_type::assign(_M_data()[__n], _CharT());
(gdb) bt
#0  std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >::_M_set_length (__n=0, this=0x555555bbc850)
    at /usr/lib/gcc/x86_64-pc-linux-gnu/11.1.0/include/g++-v11/bits/basic_string.h:217
#1  std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >::operator= (__str=..., this=0x555555bbc850)
    at /usr/lib/gcc/x86_64-pc-linux-gnu/11.1.0/include/g++-v11/bits/basic_string.h:727
#2  VarDbPkg::readDepend (this=<optimized out>, p=..., v=v@entry=0x555555bd1b90, header=...) at portage/vardbpkg.cc:363
#3  0x0000555555618ad2 in PackageTest::stringMatch (this=0x555555bba760, pkg=0x555555bc1820) at search/packagetest.cc:356
#4  0x000055555561918f in PackageTest::match (this=0x555555bba760, pkg=0x7fffffffc4e0) at search/packagetest.cc:564
#5  0x00005555556161b4 in MatchAtomTest::match (this=0x555555bc14f0, p=<optimized out>) at search/matchtree.cc:93
#6  0x0000555555602cb1 in run_eix (argc=<optimized out>, argv=<optimized out>) at eix.cc:909
#7  0x000055555556b268 in run_program (argv=0x7fffffffcb78, argc=3) at ./main/main.cc:205
#8  main (argc=3, argv=0x7fffffffcb78) at ./main/main.cc:232
```

`CXXFLAGS="-march=bdver2 -O2 -pipe -fpch-preprocess -ggdb -frecord-gcc-switches"`

**Comment by vaeth** [comment]
This should already be fixed eix-0.35.2.

**Comment by vaeth** [comment]
Closing since I assume that it is fixed; please reopen if it still breaks with the newest release
